modules/stochastic_tools/include/state_sim/model/StateSimModel.h

```diff
@@ -0,0 +1,58 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef STATESIMMODEL_H
+#define STATESIMMODEL_H
+
+#include <string>
+#include <vector>
+#include "StateSimEvTimeBased.h"
+#include "StateSimEvConditionBased.h"
+#include "StateExternalCouplingUO.h"
+#include "StateSimAllEvents.h"
+#include "StateSimAllStates.h"
+#include "StateSimAllDiagrams.h"
+#include "StateSimAllVariables.h"
+#include "StateSimAllActions.h"
```

Comment:
Try not to include headers from headers unless they are needed. In most cases you can use a "forward declaration" and then include the header from the .C file. This will drastically reduce compilation needs.